src/HaskellCI/Travis.hs

```diff
@@ -93,6 +94,10 @@ makeTravis argv config@Config {..} prj jobs@JobVersions {..} = do
 
     -- before install: we set up the environment, install GHC/cabal on OSX
     beforeInstall <- runSh $ do
+        -- Validity checks
+        when (anyGHCJS && cfgUbuntu > Bionic) $
```

Comment:
re. Yes, can we have common checks in one place, and not copy-pasted.

Reply:
I have factored out this check into a `checkConfigValidity` function, which now lives in a new `HaskellCI.Config.Validity` module.